### DISCUSSION TITLE
How to install executable script when python is managed by rye?

### DISCUSSION THREAD

**Discussion by jbulow** [astral-sh/rye#1355] [Q&A] [answered]
When rye is managing python it will make "pip install" to install to a path (~/.rye/tools/pip/bin) which is not added to the PATH when installing rye. Am I supposed to manually add that path to the PATH environment variable?

**Comment by bluss** [comment]
@jbulow with rye, it's not intended to install pip like that, I can see how that becomes very confusing. If you want to install executable scripts (apps) like for example hatch or ruff, then you'd want to use `rye tools install hatch` and so on. The tools install only ever installs isolated tools.

Note that `rye install` is just an alias for `rye tools install`. There is no user-global environment for libraries with Rye.

**Reply by jbulow** [reply]
Ok. I'm confused. :-)

How do I install a local project that I used to install with "pip install -e ." ?

When trying "rye tools install ." I get:

"error: package already installed"

But I can't use it as it is not installed somewhere to be found by PATH.

**Reply by bluss** [reply]
Already installed packages should be in rye tools list. Is the name in the list?

**Reply by jbulow** [reply]
So, my question is: When rye is managing python, how do I install things that used to work with "pip install -e ." when not using rye?

**Reply by bluss** [reply]
@jbulow depends on which environment you target for the install. If it's the user scope (pip install --user, which is often the default) then Rye does not support it.

**Reply by jbulow** [reply]
Yes, is it shown in the output of "rye tools list". But how do I run it?

**Reply by bluss** [reply]
If it's in rye tools list, then rye should have added a binary for it in .rye/shims and rye should already have ensured this directory is in the $PATH. If not, something is not correctly setup.

**Reply by jbulow** [reply]
It is not added to .rye/shims. 

What is the correct way to setup rye? I followed the instructions and selected the default options.

Is it possible to force a reinstall of a tool?

**Reply by bluss** [reply]
rye tools install -f tool

Setup sounds ok of course.

**Reply by bluss** [reply, marked as answer]
Is it possible that the tool you have installed does not have any executable scripts? If I do "rye tools install jinja2" it fits with your description - the library is listed in the list, and there are no executables. Also look at rye tools install --include-dep if that's needed, if the exectuable is from a dependency.
